### PR TITLE
Add git tag when releasing a new version

### PR DESCRIPTION
**Pull request by techtangents**

### CODE REVIEW

**Review by lnewson — APPROVED** [review]

File: src/main/ts/commands/Publish.ts [hunk 1]
```diff
@@ -50,3 +56,19 @@ export const publish = async (args: PublishArgs): Promise<void> => {
...
 };
+
+const gitTag = async (r: BranchDetails, git: gitP.SimpleGit, args: PublishArgs): Promise<void> => {
+  if (r.branchState === BranchState.ReleaseReady) {
+    const tagName = r.packageJson.name + '@' + Version.versionToString(r.version);
+    console.log(`Tagging as ${tagName}`);
+    await git.addTag(tagName);
+
+    if (args.dryRun) {
+      console.log('Dry run - not pushing tags');
```
Comment (JamesToohey) [inline src/main/ts/commands/Publish.ts:67]: Is it risky to have a tag added locally and not pushed up?

Comment (techtangents) [reply src/main/ts/commands/Publish.ts:67]: I'm not sure, to be honest.

The dry-run arg is currently described as "Don't push changes to remote systems", so I think this behaviour matches that.

I think the idea behind dry-run it to make changes locally that you can inspect and test, but not to make any dangerous changes.

Comment (JamesToohey) [reply src/main/ts/commands/Publish.ts:67]: I only ask having run into some strange things with git tags in the past. I suppose if there are any issues we'll uncover them relatively soon and just fix then.

**Review by JamesToohey — APPROVED** [review]

File: src/main/ts/commands/Publish.ts (hunk 1 above)
Comment (techtangents) [reply src/main/ts/commands/Publish.ts:67]: I don't think we'll be running dry-run much, to be honest. Cross that bridge when we get to it.

**Pull request merged**